xaitk_saliency/impls/gen_image_classifier_blackbox_sal/slidingwindow.py

```diff
@@ -0,0 +1,47 @@
+from typing import Tuple, Dict, Any
+
+import numpy as np
+from smqtk_classifier import ClassifyImage
+
+from xaitk_saliency.impls.gen_classifier_conf_sal.occlusion_scoring import OcclusionScoring
+from xaitk_saliency.impls.gen_image_classifier_blackbox_sal.occlusion_based import PerturbationOcclusion
+from xaitk_saliency.impls.perturb_image.sliding_window import SlidingWindow
+from xaitk_saliency.interfaces.gen_image_classifier_blackbox_sal import GenerateImageClassifierBlackboxSaliency
+
+
+class SlidingWindowStack (GenerateImageClassifierBlackboxSaliency):
+    """
+    Encapsulation of the perturbation-occlusion method using specifically
+    sliding windows and the occlusion-scoring method.
+    """
+
+    def __init__(
+        self,
+        window_size: Tuple[int, int] = (50, 50),
+        stride: Tuple[int, int] = (20, 20),
+        threads: int = 0
+    ):
+        self._po = PerturbationOcclusion(
+            perturber=SlidingWindow(
+                window_size=window_size,
+                stride=stride,
+            ),
+            generator=OcclusionScoring(),
+            threads=threads,
+        )
+
+    def _generate(self, ref_image: np.ndarray, blackbox: ClassifyImage) -> np.ndarray:
+        return self._po.generate(ref_image, blackbox)
+
+    @classmethod
+    def get_default_config(cls) -> Dict[str, Any]:
+        # Minor override following SlidingWindow impl because of list/tuple
+        # equality check differences.
+        cfg = super().get_default_config()
+        cfg.update(SlidingWindow.get_default_config())
+        return cfg
+
+    def get_config(self) -> Dict[str, Any]:
+        c = self._po.perturber.get_config()
+        c['threads'] = self._po.threads
```

Comment:
Why do we have to explicitly set `threads` in the config here, but not in the corresponding RISEStack method?

Reply:
This has to do with what parameters are defined in the `perturber` instance that are also "shared" with this implementation. In the RISE case, the RISE perturbation implementation had a threading parameter that is set in the `RISEStack` constructor to both that impl *and* the `PerturbationOcclusion` constructor. In this case, with `SlidingWindowStack`, the perturber being drawn from does *not* have a threading parameter, which is only encoded in the `PerturbationOcclusion` instance.